truffleruby/src/main/java/org/truffleruby/language/TruffleBootNodes.java

```diff
@@ -78,13 +78,29 @@ public RubyContext context() {
         }
     }
 
-    @CoreMethod(names = "main", onSingleton = true)
+    @CoreMethod(names = "main", onSingleton = true, optional = 1)
     public abstract static class MainNode extends CoreMethodArrayArgumentsNode {
 
         @Specialization
-        public Object main(VirtualFrame frame, @Cached("create()") IndirectCallNode callNode) {
-            String inputFile = getContext().getOriginalInputFile();
+        public Object main(
+                VirtualFrame frame,
+                NotProvided path,
```

Comment:
The current style is method arguments should be on the same line, unless they are really long, except `@Cached` which are alone on their own lines.